alc/backends/pulseaudio.cpp

```diff
@@ -891,14 +904,6 @@ void PulsePlayback::open(const char *name)
     mFrameSize = static_cast<uint>(pa_frame_size(pa_stream_get_sample_spec(mStream)));
 
     mDeviceName = pulse_name ? al::make_optional<std::string>(pulse_name) : al::nullopt;
-    if(!dev_name)
-    {
-        pa_operation *op{pa_context_get_sink_info_by_name(mContext, pulse_name,
-            &PulsePlayback::sinkNameCallbackC, this)};
-        mMainloop.waitForOperation(op, plock);
-    }
-    else
-        mDevice->DeviceName = dev_name;
```

Comment:
If there's multiple devices with the same description, OpenAL Soft appends `#2`, `#3`, etc, to them so they can be distinguished for the app to select between them. That needs to be maintained so the app sees the same device name that it used to open it, in case it wants to save/open the same device again.

Reply:
I moved it to sinkInfoCallback since it's called everytime anyway now:
https://github.com/kcat/openal-soft/pull/515/commits/013876554952153cd79f7b656814745c981192ac#diff-c73de3dc72d31b51bc4950a6d02421036e422f9707eaf72c55fdf0caf5bffc43R784

Reply:
That gets the description provided by PulseAudio, but like I said, if there happens to be multiple devices with the same description, OpenAL Soft appends `#2`, `#3`, etc, as necessary. For example, if I have two USB audio devices, and PulseAudio gives the description as `"USB Audio Analog Stereo"` for both of them, OpenAL Soft will enumerate them as:
```
USB Audio Analog Stereo
USB Audio Analog Stereo #2
```
so the desired one can still be selected/opened. If the app opens `"USB Audio Analog Stereo #2"`, the device name shouldn't be reported as `"USB Audio Analog Stereo"`. The `sinkInfoCallback` method just gets the description from PulseAudio, which doesn't account for this.

Reply:
Got it